e2e/util/validation.go

```diff
@@ -76,30 +79,48 @@ func GetRamenNameSpace(k8sClient *kubernetes.Clientset) (string, error) {
 
 // IsOpenShiftCluster checks if the given Kubernetes cluster is an OpenShift cluster.
 // It returns true if the cluster is OpenShift, false otherwise, along with any error encountered.
-func IsOpenShiftCluster(k8sClient *kubernetes.Clientset) (bool, error) {
-	discoveryClient := k8sClient.Discovery()
+func IsOpenShiftCluster(k8sClient client.Client) (bool, error) {
+	configList := &unstructured.Unstructured{}
+	configList.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "config.openshift.io",
+		Version: "v1",
+		Kind:    "ClusterVersion",
+	})
 
-	apiGroups, err := discoveryClient.ServerGroups()
-	if err != nil {
-		return false, err
+	err := k8sClient.List(context.TODO(), configList)
+	if err == nil {
+		// found OpenShift only resource type, it is OpenShift
+		return true, nil
 	}
 
-	for _, group := range apiGroups.Groups {
-		if group.Name == "route.openshift.io" {
-			return true, nil
```

Comment:
This was the same kind of check with different resource. Why `config.openshift.io` is better than `route.openshift.io`?

Reply:
My original approach was to look for routes as that is one of the earliest OpenShift additions to Kubernetes. However, config.openshift.io also been available from the very first 4.x OpenShift release at least https://docs.okd.io/4.6/rest_api/config_apis/project-config-openshift-io-v1.html.

The move to config.openshift.io is because the CR of this kind with the name `cluster` is where OpenShift stores all the cluster info. This seemed like a more appropriate way to verify OpenShift cluster than looking for routes. Between routes and config, config is more likely to stay in the feature set.